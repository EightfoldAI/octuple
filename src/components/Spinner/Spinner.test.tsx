import React from 'react';
import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import MatchMediaMock from 'jest-matchmedia-mock';
import { Spinner, SpinnerProps, SpinnerSize } from './';
import { render } from '@testing-library/react';

Enzyme.configure({ adapter: new Adapter() });

let matchMedia: any;

const spinnerProps: SpinnerProps = {
  size: SpinnerSize.Default,
  classNames: 'my-spinner-class',
};

describe('Spinner', () => {
  beforeAll(() => {
    matchMedia = new MatchMediaMock();
  });

  afterEach(() => {
    matchMedia.clear();
  });

  test('Renders without crashing', () => {
    const { container } = render(<Spinner {...spinnerProps} />);
    const spinner = container.getElementsByClassName('spinner');
    expect(() => container).not.toThrowError();
    expect(spinner).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  test('Spinner accepts custom classes', () => {
    const { container } = render(<Spinner {...spinnerProps} />);
    expect(container.querySelector('.my-spinner-class')).toBeTruthy();
  });
});
