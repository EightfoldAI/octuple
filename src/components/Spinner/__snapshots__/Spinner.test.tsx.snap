// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Renders without crashing 1`] = `
<div>
  <span
    aria-hidden="false"
    class="spinner my-spinner-class icon-wrapper"
    role="presentation"
  >
    <svg
      role="presentation"
      style="width: 48px; height: 48px;"
      viewBox="0 0 24 24"
    >
      <style>
        @keyframes spin { from { transform: rotate(0deg) } to { transform: rotate(360deg) } }
      </style>
      <g
        style="animation: spin linear 0.8s infinite; transform-origin: center;"
      >
        <path
          d="M12,4V2A10,10 0 0,0 2,12H4A8,8 0 0,1 12,4Z"
          style="fill: currentColor;"
        />
        <rect
          fill="transparent"
          height="24"
          width="24"
        />
      </g>
    </svg>
  </span>
</div>
`;
